src/commands/domain.js

```diff
@@ -1,16 +1,19 @@
-'use strict';
-
-const Application = require('../models/application.js');
-const Logger = require('../logger.js');
-const {
-  get: getApp,
-  addDomain,
-  getFavouriteDomain: getFavouriteDomainWithError,
-  markFavouriteDomain,
-  unmarkFavouriteDomain,
-  removeDomain,
-} = require('@clevercloud/client/cjs/api/v2/application.js');
-const { sendToApi } = require('../models/send-to-api.js');
+import * as Application from '../models/application.js';
+import { Logger } from '../logger.js';
+import { get as getApp, addDomain, getFavouriteDomain as getFavouriteDomainWithError, markFavouriteDomain, unmarkFavouriteDomain, removeDomain } from '@clevercloud/client/esm/api/v2/application.js';
+import { sendToApi } from '../models/send-to-api.js';
+import colors from 'colors/safe.js';
+import { parse as parseDomain } from 'tldts';
+import { diagDomainConfig } from '@clevercloud/client/esm/utils/diag-domain-config.js';
+import { NodeDnsResolver } from '../models/node-dns-resolver.js';
+
+/**
+ * @typedef {import('@clevercloud/client/esm/utils/diag-domain-config.types.js').DomainInfo} DomainInfo
+ * @typedef {import('@clevercloud/client/esm/utils/diag-domain-config.types.js').ResolveDnsResult} ResolveDnsResult
+ * @typedef {import('@clevercloud/client/esm/utils/diag-domain-config.types.js').DomainDiag} DomainDiag
+ */
+
+const resolver = new NodeDnsResolver();
```

Comment:
question: do we really want to instanciate the resolver here? Shouldn't we instantiate a resolver only when we need it?